### PR TITLE
Ins 847/ Ins 860

### PR DESCRIPTION
**Pull request by egorikas**
**- What I did**
Executor removes indexes which aren't his
**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger/artifactmanager/handler.go [hunk 1]
```diff
@@ -815,13 +834,26 @@ func (h *MessageHandler) handleHotRecords(ctx context.Context, genericMsg core.P
...
 		if err != nil {
 			inslog.Error(err)
 			continue
 		}
+
 		meta.TTL--
-		h.Recent.AddObjectWithTTL(id, meta.TTL)
+		h.Recent.AddObjectWithTLL(id, meta.TTL)
+		if savedIndex != nil {
+			err = h.Recent.MarkAsMine(id)
```
Comment (andreyromancev) [inline ledger/artifactmanager/handler.go]: No need to mark as mine here. When this handler is called, there are no records on my node.

File: ledger/pulsemanager/pulsemanager.go [hunk 2]
```diff
@@ -147,29 +147,38 @@ func (m *PulseManager) processRecentObjects(
...
-			inslogger.FromContext(ctx).Error(err)
+			logger.Error(err)
 			continue
 		}
 		recentObjects[id] = &message.HotIndex{
 			TTL:   ttl,
 			Index: encoded,
 		}
+
+		if m.Recent.IsMine(id) {
```
Comment (andreyromancev) [inline ledger/pulsemanager/pulsemanager.go]: If not mine.

File: ledger/pulsemanager/pulsemanager.go [hunk 3]
```diff
@@ -147,29 +147,38 @@ func (m *PulseManager) processRecentObjects(
...
+			err := m.db.RemoveObjectIndex(ctx, &id)
+			if err != nil {
```
Comment (andreyromancev) [inline ledger/pulsemanager/pulsemanager.go]: If we have error here, it means something wrong with DB and we have inconsistent state. It's safer to end execution. Furthermore the client has no way of knowing that something went wrong.

File: ledger/recentstorage/recentstorage.go [hunk 4]
```diff
@@ -8,11 +8,14 @@ import (
 //go:generate minimock -i github.com/insolar/insolar/ledger/recentstorage.RecentStorage -o ./ -s _mock.go
 type RecentStorage interface {
 	AddObject(id core.RecordID)
-	AddObjectWithTTL(id core.RecordID, ttl int)
+	AddObjectWithTLL(id core.RecordID, ttl int)
```
Comment (andreyromancev) [inline ledger/recentstorage/recentstorage.go]: TTL?

File: ledger/artifactmanager/handler.go [hunk 5]
```diff
@@ -439,8 +439,7 @@ func (h *MessageHandler) handleUpdateObject(ctx context.Context, pulseNumber cor
 			return errors.New("invalid state record")
 		}
 
-		h.Recent.AddObject(*msg.Object.Record())
-		err = h.Recent.MarkAsMine(*msg.Object.Record())
+		h.Recent.AddObject(*msg.Object.Record(), true)
 		if err != nil {
```
Comment (ita-sammann) [inline ledger/artifactmanager/handler.go]: err?

File: ledger/artifactmanager/handler.go [hunk 6]
```diff
@@ -463,8 +462,7 @@ func (h *MessageHandler) handleUpdateObject(ctx context.Context, pulseNumber cor
 		return nil, err
 	}
 
-	h.Recent.AddObject(*msg.Object.Record())
-	err = h.Recent.MarkAsMine(*msg.Object.Record())
+	h.Recent.AddObject(*msg.Object.Record(), true)
 	if err != nil {
```
Comment (ita-sammann) [inline ledger/artifactmanager/handler.go]: same as above

**Pull request merged**